javacord-api/src/main/java/org/javacord/api/entity/server/Server.java

```diff
@@ -1519,6 +1529,62 @@ default ServerUpdater createUpdater() {
         return createUpdater().setNickname(user, null).setAuditLogReason(reason).update();
     }
 
+    /**
+     * Timeout the given user on this server.
+     *
+     * <p>If you want to update several settings at once, it's recommended to use the
+     * {@link ServerUpdater} from {@link #createUpdater()} which provides a better performance!
+     *
+     * @param user    The user.
+     * @param timeout The timestamp until the user should be timeout,
+     * @return A future to check if the update was successful.
+     */
+    default CompletableFuture<Void> timeoutUser(User user, Instant timeout) {
+        return createUpdater().setUserTimeout(user, timeout).update();
+    }
+
+    /**
+     * Timeout the given user on this server.
+     *
+     * <p>If you want to update several settings at once, it's recommended to use the
+     * {@link ServerUpdater} from {@link #createUpdater()} which provides a better performance!
+     *
+     * @param user    The user.
+     * @param timeout The timestamp until the user should be timeout,
+     * @param reason  The audit log reason for this update.
+     * @return A future to check if the update was successful.
+     */
+    default CompletableFuture<Void> timeoutUser(User user, Instant timeout, String reason) {
+        return createUpdater().setUserTimeout(user, timeout).setAuditLogReason(reason).update();
+    }
+
```

Comment:
What do you think of also adding convenience methods with a `User, Duration` signature to make "timeout this user for 48 hours starting now" style calls easier?

```suggestion

    /**
     * Timeout the given user on this server.
     *
     * <p>If you want to update several settings at once, it's recommended to use the
     * {@link ServerUpdater} from {@link #createUpdater()} which provides a better performance!
     *
     * @param user    The user.
     * @param duration The duration how long the user should be timed out.
     * @param reason  The audit log reason for this update.
     * @return A future to check if the update was successful.
     */
    default CompletableFuture<Void> timeoutUser(User user, Duration timeout, String reason) {
        return createUpdater().setUserTimeout(user, Instant.now().plus(duration)).setAuditLogReason(reason).update();
    }
```

Reply:
Personally I think the use case would be minimal and just clutter the API. I can imagine the use case of timeouts would be to integrate it into moderation bots where mods can timeout a user with a command `/timeout Domme 12` and then the convenience methods are useless. And the time range is just so huge (+28days) so what would be the most used times to make convenience methods for?

Reply:
I think Saladocs suggestion (the code above) is a good idea. With your current implementation, you need to calculate the target instant / timestamp yourself.
With the suggest additional method, you can go like
```java
server.timeoutUser(user, Duration.ofHours(12));
```

Reply:
Oh, I totally misunderstood what he meant. Indeed, this is a good idea